### ISSUE TITLE
Deploying helm chart in cluster with custom cluster domain will result in failure to start

### ISSUE THREAD

**Issue by steve-todorov** [gitpod-io/gitpod#1759]
### Describe the bug

When you attempt to deploy gitpod in a kubernetes cluster which has a custom cluster domain the `proxy` pod will fail to start with the following error:

```
Starting nginx
2020/08/27 19:22:19 [emerg] 1#1: host not found in upstream "server.gitpod.svc.cluster.local:3000" in /etc/nginx/vhost.upstreams.conf:3
nginx: [emerg] host not found in upstream "server.gitpod.svc.cluster.local:3000" in /etc/nginx/vhost.upstreams.conf:3
```

The error is caused by the hard-coded domain in `chart/config/proxy/vhost.upstreams.conf` https://github.com/gitpod-io/gitpod/blob/609bd8cd554d32f50f65886137efc2fe1343771e/chart/config/proxy/vhost.upstreams.conf#L1-L37

It seems like this is also hard-coded in other files under the nginx configuration dir, but I haven't gone through all of them so unfortunately I can't list them here.

### Example `values.yml`

```
---
hostname: "gitpod-example.domain.com"
defaults:
  dnsPolicy: ClusterFirst
  restartPolicy: Always
  terminationGracePeriodSeconds: 30
```

**Comment by geropl** [comment]
Hi @steve-todorov ,

thx for bringing this up. We're currently (at this very moment!) working to improve the docs for the "bring-your-own-domain" case!

**Comment by geropl** [comment]
Hi @steve-todorov , 

sorry for the long delay. We finally released a new version of self-hosted, `0.6.0`.
The docs are updated accordingly: https://www.gitpod.io/docs/self-hosted/latest/self-hosted/

Please give it a try and re-open if the error persists! :pray:

**Comment by soundsannoying** [comment]
@geropl I have the same issue as steve. I deployed GitPod 0.6.0 with a custom hostname as described in the docs. And the proxy pod keeps crashing over and over again with the following error message:
[emerg] 1#1: host not found in upstream "server.oppsee-space.svc.cluster.local:3000" in /etc/nginx/vhost.upstreams.conf:3
nginx: [emerg] host not found in upstream "server.oppsee-space.svc.cluster.local:3000" in /etc/nginx/vhost.upstreams.conf:3
